packages/typespec-ts/src/modular/emitSamples.ts

```diff
@@ -0,0 +1,516 @@
+  dpgContext: SdkContext,
+  method: SdkServiceMethod<SdkServiceOperation>,
+  parameterMap: Record<string, SdkHttpParameterExample>,
+  topLevelClient: SdkClientType<SdkServiceOperation>
+): ExampleValue[] {
+  // TODO: blocked by TCGC issue: https://github.com/Azure/typespec-azure/issues/1419
+  // refine this to support generic client-level parameters once resolved
+  const result: ExampleValue[] = [];
+  const credentialExampleValue = getCredentialExampleValue(
+    dpgContext,
+    topLevelClient.initialization
+  );
+  if (credentialExampleValue) {
+    result.push(credentialExampleValue);
+  }
+
+  let subscriptionIdValue = `"00000000-0000-0000-0000-00000000000"`;
+  // required client-level parameters
+  for (const param of method.operation.parameters) {
+    if (
+      param.onClient === false ||
+      param.optional === true ||
+      param.type.kind === "constant"
+    ) {
+      continue;
+    }
+
+    const exampleValue = parameterMap[param.serializedName];
+    if (!exampleValue || !exampleValue.value) {
+      // report diagnostic if required parameter is missing
+      reportDiagnostic(dpgContext.program, {
+        code: "required-sample-parameter",
+        format: {
+          exampleName: method.name,
+          paramName: param.name
+        },
+        target: NoTarget
+      });
+      continue;
+    }
+
+    if (
+      param.name.toLowerCase() === "subscriptionid" &&
+      isArm(dpgContext) &&
+      exampleValue
+    ) {
+      subscriptionIdValue = getParameterValue(exampleValue.value);
+      continue;
+    }
+    result.push({
+      name: exampleValue.parameter.name,
+      value: getParameterValue(exampleValue.value),
+      isOptional: Boolean(param.optional)
+    });
+  }
+  // always add subscriptionId for ARM clients
+  if (isArm(dpgContext)) {
+    result.push({
+      name: "subscriptionId",
+      value: subscriptionIdValue,
+      isOptional: false
+    });
+  }
+  // optional parameters
+  method.operation.parameters
+    .filter(
+      (param) =>
+        param.onClient === true &&
+        param.optional === true &&
+        parameterMap[param.serializedName]
+    )
+    .map((param) => parameterMap[param.serializedName]!)
+    .forEach((param) => {
+      result.push({
+        name: param.parameter.name,
+        value: getParameterValue(param.value),
+        isOptional: true
+      });
+    });
+
+  return result;
+}
+
+function getCredentialExampleValue(
+  dpgContext: SdkContext,
+  initialization: SdkInitializationType
+): ExampleValue | undefined {
+  const keyCredential = hasKeyCredential(initialization),
+    tokenCredential = hasTokenCredential(initialization);
+  if (keyCredential || tokenCredential) {
+    if (isAzurePackage({ options: dpgContext.rlcOptions })) {
+      // Support DefaultAzureCredential for Azure packages
+      return {
+        name: "credential",
+        value: `new ${resolveReference(
+          AzureIdentityDependencies.DefaultAzureCredential
```

Comment:
I didn't see any changes related to set the azure/identity dependency in the package.json? I wonder why it's there if sample generation is not working before?

Reply:
we would include this automatically in sample's package.json file: https://github.com/Azure/azure-sdk-for-js/blob/main/sdk/compute/arm-compute/samples/v22/typescript/package.json#L34 .